tests/helpers_tests/test_condition.py

```diff
@@ -108,6 +126,273 @@ def test_evaluate__returns_null__when_condition_has_an_invalid_type_property(sel
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_condition_list, {'Android': '2'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_2_matches_target_version_2(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_condition_list, {'Android': '2.2'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_0_matches_target_version_2_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_0_condition_list, {'Android': '2.0'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_0_0_matches_target_version_2_0_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_0_0_condition_list, {'Android': '2.0.0'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_false__when_user_version_2_0_does_not_match_target_version_2_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_equal_2_0_0_condition_list, {'Android': '2.0'}, self.mock_client_logger
+        )
+
+        self.assertStrictFalse(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_0_0_release_is_greater_than_target_version_2_0_0_beta(
+            self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_greater_than_2_0_0_beta_condition_list, {'Android': '2.0.0-release'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_2_0_1_is_greater_than_target_version__2_0_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_greater_than_2_0_0_condition_list, {'Android': '2.0.1'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_1_9_9_is_less_than_target_version_2_0_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_less_than_2_0_0_condition_list, {'Android': '1.9.9'}, self.mock_client_logger
+        )
+
+        self.assertStrictTrue(evaluator.evaluate(0))
+
+    def test_evaluate__returns_true__when_user_version_1_9_0_beta_is_less_than_target_version_2_0_0(self):
+
+        evaluator = condition_helper.CustomAttributeConditionEvaluator(
+            semver_less_than_2_0_0_condition_list, {'Android': '1.9.0-beta'}, self.mock_client_logger
```

Comment:
I think it will get better to compare 2.0.0-beta against 2.0.0

Reply:
How is this test different from test_evaluate__returns_true__when_user_version_2_0_0_beta_is_less_than_target_version_2_0_0_release ? We are not learning anything new.

Reply:
Yeah, removed this one. The other that you mentioned should suffice.